src/Components/src/Microsoft.AspNetCore.Components.Server/Circuits/RemoteRenderer.cs

```diff
@@ -90,6 +90,36 @@ public void AddComponent(Type componentType, string domElementSelector)
             RenderRootComponent(componentId);
         }
 
+        /// <inheritdoc />
+        public override Task Invoke(Action workItem)
+        {
+            if (SynchronizationContext.Current == _syncContext)
+            {
+                // No need to dispatch. Avoid deadlock by invoking directly.
+                return base.Invoke(workItem);
```

Comment:
@rynowak Do you agree this is the right thing to do? If we don't, developers will have to code their own checks about whether to dispatch or not.

Reply:
yep.